src/System.Private.CoreLib/src/System/DateTime.WinRT.cs

```diff
@@ -0,0 +1,20 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System
+{
+    public readonly partial struct DateTime
+    {
+        private static unsafe bool SystemSupportsLeapSeconds()
+        {
+            long l = 0;
+
+            return Interop.Kernel32.GetProcessInformation(
```

Comment:
Is this really how we check whether leap seconds are supported in an appcontainer? (I.e. we don't look at the returned `PROCESS_LEAP_SECOND_INFO` structure, but return whether the `GetProcessInformation` API call succeeded?

Also, why don't we just use this same mechanism for Win32, but instead call into an API that is documented as "subject to change, do not call" (`[NtQuerySystemInformation may be altered or unavailable in future versions of Windows. Applications should use the alternate functions listed in this topic.]` - [reference](https://docs.microsoft.com/en-us/windows/desktop/api/winternl/nf-winternl-ntquerysysteminformation))

Reply:
I have fixed the interop to use proper `PROCESS_LEAP_SECOND_INFO` structure. (I am not sure whether Tarek is around this week.)

I know Tarek worked with Windows folks on this change, so there is probably a good reason why he choose to use `NtQuerySystemInformation`. He can comment on it once he is around.

Reply:
> I have fixed the interop to use proper PROCESS_LEAP_SECOND_INFO structure. (I am not sure whether Tarek is around this week.)

@jkotas Your change is ok, but not really needed. there was no real need to define PROCESS_LEAP_SECOND_INFO. 
And thanks for your follow up to merge this one.

Reply:
> there was no real need to define PROCESS_LEAP_SECOND_INFO.

Why not? The public signature of the function has `PROCESS_LEAP_SECOND_INFO`, so that's what we should use in our interop wrappers. We do not use cryptic shortcuts in our interop definitions.

Reply:
> Your change is ok

@tarekgh Are you sure about this? My change looks at `PROCESS_LEAP_SECOND_INFO` - that you are saying is not desirable in your other reply.

I do not understand why we would want to alter our behavior for apps that have not opted into this.

Reply:
> My change looks at PROCESS_LEAP_SECOND_INFO - that you are saying is not desirable in your other reply.

Sorry I missed check 

```C#
info.Flags & Interop.Kernel32.PROCESS_LEAP_SECOND_INFO_FLAG_ENABLE_SIXTY_SECOND) != 0;
```

This is wrong and need to be removed.

In the systems support leap seconds, the leap seconds is going to be inserted in the system anyway either the app opt-in to get leap second or not. the apps opt-in get the leap seconds will get a second 60 when the leap second occur. the app opt-out (which is the default behavior), the app will get the second 59 twice when the leap second occur. in both cases, the leap seconds is counted in the system and we need to handle that.

Reply:
@jkotas Just checking, do you want me fix this? or you'll fix it.

Reply:
Jan's change just mirrors what the code is doing in the .Win32 version - does it need to be fixed there too?  (That's why I raised this initially - there was an inconsistency.)

Reply:
@tarekgh It would be great if you can fix it. Thanks!

> In the systems support leap seconds...

The matrix of different behaviors makes my head explode...

Reply:
I have submitted a PR and CC'ed both of you. here is some more information I hope it'll clarify more about the behavior

```C#
if (RS5+ System)
{
    // NtQuerySystemInformation(SystemLeapSecondInformation) should succeed and fill SYSTEM_LEAP_SECOND_INFORMATION struct.
    //      SYSTEM_LEAP_SECOND_INFORMATION.Enabled will tell if the leap seconds is globally enabled or not.
    //      if (SYSTEM_LEAP_SECOND_INFORMATION.Enabled == false)
    //          The system behave as if RS4 and below (no leap seconds support at all). system admins can do that.
    //      We cannot call NtQuerySystemInformation on UWP app containers.
    //
    //  GetProcessInformation(ProcessLeapSecondInfo) should succeed and fill PROCESS_LEAP_SECOND_INFO struct
    //      if (PROCESS_LEAP_SECOND_INFO.Flags & PROCESS_LEAP_SECOND_INFO_FLAG_ENABLE_SIXTY_SECOND)
    //          The app opted-in to get the leap second as 60 value (For example: 23:59:59 -> 23:59:60 -> 00:00:00)
    //      else
    //          The app opted-out to get the leap second as 60 value (For example: 23:59:59 -> 23:59:59 -> 00:00:00)
    //      Note, in either cases the leap seconds in the system is counted.
    //      We can call GetProcessInformation on UWP app containers.
    //
    //   if the system admin disabled the leapseconds system wide, GetProcessInformation(ProcessLeapSecondInfo) will still succeed
    //   at that time (in UWP container apps) we still feel the system support leap seconds but we still behave correctly because 
    //   all information we get from the system will work in our calculations.we'll just pay the extra perf cost we are paying for handling leap seconds.
    //   We have requested a new API like NtQuerySystemInformation to be used in UWP containers apps to replace using
    //   GetProcessInformation
    //
}
else
{
    // NtQuerySystemInformation(SystemLeapSecondInformation) and GetProcessInformation(ProcessLeapSecondInfo) calls will fail
}
```